slsDetectorGui/src/qDetectorMain.cpp

```diff
@@ -242,30 +241,27 @@ void qDetectorMain::SetUpDetector(const std::string fName, int multiID) {
         LoadConfigFile(fName);
 
     // validate detector type (for GUI) and update menu
-    detType = myDet->getDetectorTypeAsEnum();
+    detType = det->getDetectorType().tsquash("Different detector type for all modules.");
+    actionTrimbitsLoad->setEnabled(false);
     switch (detType) {
     case slsDetectorDefs::EIGER:
+        actionTrimbitsLoad->setEnabled(true);
         break;
     case slsDetectorDefs::GOTTHARD:
     case slsDetectorDefs::JUNGFRAU:
-        actionLoadTrimbits->setText("Load Settings");
-        actionSaveTrimbits->setText("Save Settings");
-        break;
     case slsDetectorDefs::MOENCH:
-        actionLoadTrimbits->setEnabled(false);
-        actionSaveTrimbits->setEnabled(false);
-        break;
+         break;
     default:
         std::string errorMess =
-            myDet->getHostname() + std::string(" has ") +
-            myDet->getDetectorTypeAsString() + std::string(" detector type (") +
+            sls::ToString(det->getHostname(), ',') + std::string(" has ") +
```

Comment:
This should also work without the std::string() around the string litterals if at least one object is a string. 

 sls::ToString(det->getHostname(), ',') +  has " + ...

should work and is a bit cleaner.

Reply:
done